src/wechaty-puppet/puppet.go

```diff
@@ -1,9 +1,66 @@
 package wechaty_puppet
 
 import (
+	lru "github.com/hashicorp/golang-lru"
+
 	"github.com/wechaty/go-wechaty/src/wechaty-puppet/schemas"
 )
 
-type Puppet interface {
+type PuppetInterface interface {
```

Comment:
Q3: In my understanding, Puppet should be an interface definition, different protocols to implement different classes, but some methods of Puppet have logic, and they cannot have logic in Go's interface definition. I don't know what to do.

Sorry, maybe my understanding of wechaty and wechat-puppet is too simple, and I may need to read and understand them as a whole.